### DISCUSSION TITLE
Using WebAuthn with Supabase

### DISCUSSION THREAD

**Discussion by danielpox** [supabase/supabase#9216] [Questions] [answered]
If I want to enable WebAuthn for my web app (to use Passkeys instead of passwords), how would this work with the existing auth infrastructure in Supabase?

Do I have to manually store the user’s public keys in a database table? Or could a more seamless integration be considered to be built-in into Supabase?

**Comment by J0** [comment, marked as answer]
Hey @danielpox,

Unfortunately, we don't have first class support for WebAuthn as a sign-in replacement in our auth infrastructure at the moment so you would have to store the public keys in the database table. We are, however, planning to introduce WebAuthn as a second factor for Multi-Factor Authentication. For the moment, you may want to consider Magiclink as a passwordless alternative

If you'd like to see the feature, please feel free to express interest [in the existing thread](https://github.com/supabase/gotrue/issues/92)  by upvoting/commenting with your use case.

Let us k

Sorry about this!

**Comment by Manc** [comment]
Has anyone solved this yet? I'm new to Supabase and I feel like I'm only missing one tiny piece of the puzzle, which is actually signing the user in, once my custom server-side code is confident that the user should now be signed in.

So far, I have set up user authentication with magic email link and a signed-in user is able to register an authenticator, which is stored in a custom DB table with its public key. When the user is not signed in, they can authenticate with WebAuthn, challenge is verified etc., and everything is ready to start a Supabase session and set the cookie, but I cannot find a way to manually start a session from the server.

I'm looking for something like this:

```TS
const supabaseServerClient = createServerSupabaseClient({ req, res });
const signinResponse = await supabaseServerClient.auth.signIn({ email }); // `signIn` does not exist
```

Is there any way to log in a user with just their user ID or email address? Or do I have to roll my own session management etc. and cannot use any of the Supabase features, including row level security? Thanks for any hints!

**Reply by RihanArfan** [reply]
See https://supabase.com/docs/guides/auth/auth-magic-link

**Comment by zdarin** [comment]
If someone interested in Passkey here basic setup by @oouazize and reduced code for basic integrations by me @zdarin 
<br> [Supabase, Simplewebauthn for Passkey, Nextjs 15, Tailwind v4, Shadcn](https://github.com/zdarin/supabase-passkey-simplewebauthn)
